[Console] Add option to disable keyboard shortcuts

mibragimov opened the pull request:
Closes https://github.com/elastic/kibana/issues/48410

### Summary

This PR adds support for disabling keyboard shortcuts in Console. Shortcuts can be enabled/disabled in the settings modal. 


<img width="359" alt="Capture" src="https://user-images.githubusercontent.com/53621505/160841804-e8bcd9bf-0b69-4774-b614-a297d89c102a.PNG">

sabarasaba reviewed (commented):
Thanks for having a go at this @mibragimov! I've had a look at the code and left a few questions, tested locally and works just fine!

mibragimov commented:
@elasticmachine merge upstream

kibanamachine commented:
expected head sha didn’t match current head ref.

elasticmachine commented:
Pinging @elastic/platform-deployment-management (Team:Deployment Management)

mibragimov commented:
@elasticmachine merge upstream

mibragimov commented on src/plugins/console/public/application/containers/editor/legacy/console_editor/keyboard_shortcuts.ts:
```diff
@@ -28,39 +28,44 @@ export function registerCommands({
   coreEditor.registerKeyboardShortcut({
     keys: { win: 'Ctrl-Enter', mac: 'Command-Enter' },
-    name: 'send to Elasticsearch',
+    name: '__console_send_to_Elasticsearch',
```
@sabarasaba I've changed the implementation a bit. Instead of prefixing the every command with `__console`, what if we define the commands that need to be disabled in `unregiterKeyboardShortcuts`. See [here](https://github.com/elastic/kibana/pull/128887/files#diff-ca1f078bd63c665b23b0cadf62c1bc384efb2a8039169aaa5ca6a483a0e28bffR83). WDYT?

sabarasaba replied:
love it, i think it s a lot clearer now!

sabarasaba commented on src/plugins/console/public/application/models/legacy_core_editor/legacy_core_editor.ts:
```diff
@@ -297,6 +297,11 @@ export class LegacyCoreEditor implements CoreEditor {
   }
 
+  unregisterKeyboardShortcut(command: string) {
+    // @ts-ignore
```
why do we need to ignore this next line? 🤔

mibragimov replied:
Getting the following ts error here ` Property 'removeCommand' does not exist on type 'CommandManager'` but we have access to it.

sabarasaba commented on src/plugins/console/public/services/settings.ts:
```diff
@@ -30,72 +31,96 @@ export interface DevToolsSettings {
+  isKeyboardShortcutsDisabled: boolean;
+}
+
+enum SettingKeys {
```
love this!

sabarasaba reviewed (approved):
Thanks for addressing all the feedback @mibragimov! I've re-tested everything and works fine, left a comment about a line that we're ignoring linting from. But I'm gonna go ahead and approve it for now and wait for clarification on that.

sabarasaba commented:
Also `v8.2.0` is past feature freeze, so we might want to use `v8.3.0` instead.

alexwizp commented on src/plugins/console/public/application/components/settings_modal.tsx:
```diff
@@ -267,15 +284,34 @@ export function DevToolsSettingsModal(props: Props) {
               />
             }
-            onChange={(e) => setHistoryDisabled(e.target.checked)}
+            onChange={(e) => setIsHistoryDisabled(e.target.checked)}
```
nit: please wrap it in `useCallback` to avoid extra rendering.

alexwizp commented on src/plugins/console/public/application/containers/editor/legacy/console_editor/keyboard_shortcuts.ts:
```diff
@@ -63,4 +65,30 @@ export function registerCommands({
+    'open documentation',
+    'auto indent request',
+    'move to previous request start or end',
+    'move to next request start or end',
```
nit: any ideas about moving it into constants? any then use them both for `register` and `unregister`.

alexwizp commented on src/plugins/console/public/application/containers/main/main.tsx:
```diff
@@ -108,7 +111,7 @@ export function Main() {
         {showingHistory ? <EuiFlexItem grow={false}>{renderConsoleHistory()}</EuiFlexItem> : null}
         <EuiFlexItem>
-          <Editor loading={!done} />
+          <Editor loading={!done} setEditorInstance={(instance) => setEditorInstance(instance)} />
```
setEditorInstance={setEditorInstance}

alexwizp commented on src/plugins/console/public/services/settings.ts:
```diff
@@ -30,72 +31,96 @@ export interface DevToolsSettings {
   setFontSize(size: number) {
-    this.storage.set('font_size', size);
+    this.storage.set(SettingKeys.FONT_SIZE, size);
     return true;
```
not related to your PR, but why all setters return `true` ?

mibragimov replied:
As far as I know it may throw an exception if the storage is full

mibragimov commented:
@sabarasaba @alexwizp Thanks for the review! I've addressed the comments with [1470c91](https://github.com/elastic/kibana/pull/128887/commits/1470c913054216deb765f9abf87057a7226e4301)
